### DISCUSSION TITLE
How do you rename a subdirectory?

### DISCUSSION THREAD

**Discussion by iWantToKeepAnon** [magit/magit#5247] [Ideas]
I see `magit-file-rename` but can't find `magit-directory-rename`.  Web searching suggests there isn't one.  I have to do a `git mv orig dest` to rename a folder. ?

**Comment by kyleam** [comment]
> I have to do a `git mv orig dest` to rename a folder. ?

That'd be one way (e.g., via `!`/`magit-run`).

Or you could rename it outside of Magit, e.g., with `rename-file` or in dired and then stage the changes in Magit.

**Reply by tarsius** [reply]
Obviously we shouldn't add `magit-directory-rename`, but it would probably make sense for `magit-file-rename` to support renaming directories. Anyway, given that the mentioned, obvious workarounds exist, this is very low priority.

**Reply by iWantToKeepAnon** [reply]
> Or you could rename it outside of Magit, e.g., with `rename-file` or in dired and then stage the changes in Magit.

That would likely delete the old directory from the repo and add brand new files to the repo.  Git wouldn't know about the rename.  I saw `magit-run` but that makes the magit porcelain leaky IMO as this is a frequently used (at least by me) git command.

Low priority is fine; there are many work-arounds.

**Reply by tarsius** [reply]
> That would likely delete the old directory from the repo and add brand new files to the repo. Git wouldn't know about the rename.

In a second step you would stage both the removals and the additions, and Git would then figure out that taken together that amounts to a rename.

Note that Git does not explicitly support renames. The effect of `git mv a b` and `mv a b; git add a b` is exactly the same. Git figures out whether something is to be considered a rename, when you ask for it (and there arguments you can use to specify how similar two "files" have to be considered "the same"), but it does not record whether something "is" a rename.

**Reply by tarsius** [reply]
`magit-file-rename` now supports directories. 7ccff374d06df833a03bfd19bbce80e29833d29b
